pkg/apis/stork/v1alpha1/types.go

```diff
@@ -212,3 +217,193 @@ type MigrationList struct {
+	// Storage details of storage used by the driver
+	Storage StorageSpec `json:"storage"`
+	// CloudStorage details of storage in cloud environment.
+	// Keep this only at cluster level for now until support to change
+	// cloud storage at level is added.
+	CloudStorage CloudStorageSpec `json:"cloudStorage"`
+	// SecretsProvider is the name of secret provider that driver will connect to
+	SecretsProvider string `json:"secretsProvider"`
+	// Env is a list of environment variables used by the driver
+	Env []core.EnvVar `json:"env"`
+	// CSIEndpoint is the csi endpoint for the driver
+	CSIEndpoint string `json:"csiEndpoint"`
+	// CallHome send cluster information for analytics
+	CallHome bool `json:"callHome"`
+	// RuntimeOpts is a map of options with extra configs for storage driver
+	RuntimeOpts map[string]string `json:"runtimeOptions"`
+	// Nodes node level configurations that will override the ones at cluster
+	// level. These configurations can be grouped based on label selectors.
+	Nodes []NodeSpec `json:"nodes"`
+}
+
+// NodeSpec is the spec used to define node level configuration. Values
+// here will override the ones present at cluster-level for nodes matching
+// the selector.
+type NodeSpec struct {
+	// Selector rest of the attributes are applied to a node that matches
+	// the selector
+	Selector NodeSelector `json:"selector"`
+	// Network is the network information for storage driver
+	Network NetworkSpec `json:"network"`
+	// Storage details of storage used by the driver
+	Storage StorageSpec `json:"storage"`
+	// Geo is topology information for the node
+	Geo Geography `json:"geography"`
+	// Env is a list of environment variables used by the driver
+	Env []core.EnvVar `json:"env"`
+	// RuntimeOpts is a map of options with extra configs for storage driver
+	RuntimeOpts map[string]string `json:"runtimeOptions"`
+}
+
+// NodeSelector let's the user select a node or group of nodes based on either
+// the NodeName or the node LabelSelector. If NodeName is specified then,
+// LabelSelector is ignored as that is more precise, even though it does not
+// match any node names.
+type NodeSelector struct {
+	// NodeName is the name of Kubernetes node that it to be selected
+	NodeName string `json:"nodeName"`
+	// LabelSelector is label query over all the nodes in the cluster
+	LabelSelector meta.LabelSelector `json:"labelSelector"`
+}
+
+// KvdbSpec contains the details to access kvdb
+type KvdbSpec struct {
+	// Internal flag indicates whether to use internal kvdb or an external one
+	Internal bool `json:"internal"`
+	// Endpoints to access the kvdb
+	Endpoints []string `json:"endpoints"`
+	// BasicAuthSecret is name of kubernetes secret containing the username
+	// and password for basic auth
+	BasicAuthSecret string `json:"accessSecret"`
+	// CertificateSecret is name of the kubernetes secret containing the
+	// cert files required for kvdb authentication
+	CertificateSecret string `json:"certificateSecret"`
```

Comment:
Do we assume the certificate filenames?

Reply:
Yes. We can doc it.

Reply:
That should be on the implementation to figure out from the secret.

Reply:
How would we know what is the key that is used for client cert or CA cert? 
By default when you create a k8s secret it  takes the name of the file as key. We need to document and make sure people use a known names for different secret keys.